tests/Content/VersionTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\App;
+use Kirby\Cms\Page;
+use Kirby\Filesystem\Dir;
+use Kirby\TestCase;
+
+/**
+ * @coversDefaultClass Kirby\Content\Version
+ */
+class VersionTest extends TestCase
+{
+	public const TMP = KIRBY_TMP_DIR . '/Content.Version';
+
+	protected $model;
+
+	public function setUp(): void
+	{
+		Dir::make(static::TMP);
+
+		$this->model = new Page([
+			'kirby'    => new App(),
+			'root'     => static::TMP,
+			'slug'     => 'a-page',
+			'template' => 'article'
+		]);
+	}
+
+	public function tearDown(): void
+	{
+		App::destroy();
+		Dir::remove(static::TMP);
+	}
+
+	/**
+	 * @covers ::create
+	 */
+	public function testCreate(): void
+	{
+		$version = new Version(
+			model: $this->model,
+			id: VersionId::published()
+		);
+
+		$this->assertFalse($version->exists());
```

Comment:
These kinds of checks make the tests depend on methods of the class that aren't tested. This could result in false-positive test results if e.g. the `exists()` method uses an incorrect data source that happens to lead to the correct results in our test cases.

With the current simple implementation (mostly pass-throughs to the storage handler), a real-world bug is pretty unlikely. However it could make the tests more meaningful to use more low-level checks e.g. with `assertFileExists()`/`assertFileNotExists()` that actually verify what happens.

For the `testExists()` that applies the other way around as well.

Reply:
You are totally right about this. I would love to fix this in the version-class-improvements branch though because I've refactored the unit tests there to properly test single and multi language installations side by side. 

I've tried to already refactor the unit tests in this branch to check for existing files, but there's no proper way to do that for the multi-lang checks without those changes from the later branch.

Reply:
Added it to the todo list.